src/parsers/power_system_table_data.jl

```diff
@@ -729,9 +730,9 @@ function make_generator(data::PowerSystemTableData, gen, cost_colnames, bus)
         generator = make_thermal_generator(data, gen, cost_colnames, bus)
     elseif gen_type == ThermalMultiStart
         generator = make_thermal_generator_multistart(data, gen, cost_colnames, bus)
-    elseif gen_type <: HydroGen
+    elseif !isnothing(gen_type) && gen_type <: HydroGen
```

Comment:
This would be simpler (and make future additions easier) if you handled `isnothing(gen_type)` explicitly.